UI/CombatReport/GraphicalSummary.cpp

```diff
@@ -597,22 +669,42 @@ class OptionsBar : public GG::Wnd {
 GraphicalSummaryWnd::GraphicalSummaryWnd() :
     GG::Wnd(GG::X0, GG::Y0, GG::X1, GG::Y1, GG::INTERACTIVE),
     m_sizer(0),
-    m_options_bar(0){
+    m_options_bar(0)
+{ }
+
+GG::Pt GraphicalSummaryWnd::MinUsableSize() const {
+    GG::Pt min_size(GG::X0, GG::Y0);
+
+    // Ask the bar sizer for the required size to display the side bars because
+    // it contains all of the useful sizing information in the first place,
+    // even though it does not derive from GG::Wnd and have a virtual
+    // MinUsableSize function.
+    if(m_sizer) {
+        min_size += m_sizer->GetMinSize();
     }
 
+    if(m_options_bar) {
+        GG::Pt options_bar_min_size(m_options_bar->MinUsableSize());
+        min_size.x = std::max(min_size.x, options_bar_min_size.x);
+        min_size.y += options_bar_min_size.y;
+    }
+
+    return min_size;
+}
+
 void GraphicalSummaryWnd::SetLog(int log_id) {
     MakeSummaries(log_id);
 }
 
 void GraphicalSummaryWnd::DoLayout() {
-    GG::Pt ul(GG::X0, SIDE_BOX_MARGIN);
-
-    if(!m_options_bar) {
-        m_options_bar = new OptionsBar(m_sizer);
-        AttachChild(m_options_bar);
-        GG::Connect( m_options_bar->ChangedSignal, &GraphicalSummaryWnd::DoLayout, this );
+    if(!m_sizer) {
+        ErrorLogger() << "GraphicalSummaryWnd::DoLayout() called before "
+                         "creating m_sizer.";
```

Comment:
I try to keep lines under 80 columns and it's easy to do with string literals, I don't see any reason not to?